src/uu/cp/src/cp.rs

```diff
@@ -1865,6 +1875,53 @@ fn copy_file(
     Ok(())
 }
 
+#[cfg(unix)]
+fn handling_no_preserve_mode(options: &Options, org_mode: u32) -> u32 {
+    let (is_preserve_mode, is_explicit_no_preserve_mode) = options.preserve_mode();
+    if !is_preserve_mode {
+        use libc::{
+            S_IRGRP, S_IROTH, S_IRUSR, S_IRWXG, S_IRWXO, S_IRWXU, S_IWGRP, S_IWOTH, S_IWUSR,
+        };
+
+        #[cfg(not(any(
+            target_os = "android",
+            target_os = "macos",
+            target_os = "macos-12",
+            target_os = "freebsd",
+        )))]
+        const MODE_RW_UGO: u32 = S_IRUSR | S_IWUSR | S_IRGRP | S_IWGRP | S_IROTH | S_IWOTH;
+        #[cfg(not(any(
+            target_os = "android",
+            target_os = "macos",
+            target_os = "macos-12",
+            target_os = "freebsd",
+        )))]
+        const S_IRWXUGO: u32 = S_IRWXU | S_IRWXG | S_IRWXO;
```

Comment:
you can reduce duplication also by doing:
```suggestion
        #[cfg(not(any(
            target_os = "android",
            target_os = "macos",
            target_os = "macos-12",
            target_os = "freebsd",
        )))]
        {
             const MODE_RW_UGO: u32 = S_IRUSR | S_IWUSR | S_IRGRP | S_IWGRP | S_IROTH | S_IWOTH;
            const S_IRWXUGO: u32 = S_IRWXU | S_IRWXG | S_IRWXO;
        }
```